internal/reconciliations/istio/installation.go

```diff
@@ -100,7 +102,23 @@ func (i *Installation) Reconcile(ctx context.Context, client client.Client, isti
 			return istioCR, err
 		}
 
-		err := i.Client.Uninstall(ctx)
+		istioResourceFinder, err := resources.NewIstioResourcesFinderFromConfigYaml(ctx, client, ctrl.Log, istioResourceListPath)
+		if err != nil {
+			return istioCR, err
+		}
+
+		clientResources, err := istioResourceFinder.FindUserCreatedIstioResources()
+		if err != nil {
+			return istioCR, err
+		}
+		if len(clientResources) > 0 {
+			errorTemplate := "Could not delete Istio module instance since there are customer created resources present:"
+			for _, resource := range clientResources {
+				errorTemplate += fmt.Sprintf("\n%s:%s/%s", resource.GVK.Kind, resource.Namespace, resource.Name)
```

Comment:
listing all recourses may create a big list on big clusters. although it will be helpful, I would suggest to limit it just kind of resources (and eventually count) rather to list them all